implementations/rust/ockam/ockam_command/src/rendezvous_server/start/healthcheck.rs

```diff
@@ -0,0 +1,149 @@
+use ockam::transport::HostnamePort;
+use ockam::udp::{RendezvousClient, UdpBindArguments, UdpBindOptions, UdpTransport};
+use ockam_api::DefaultAddress;
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{route, AsyncTryClone, Error, Result};
+use ockam_node::Context;
+use std::net::SocketAddr;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpListener;
+use tokio::task::JoinHandle;
+use tracing::info;
+
+pub struct Healthcheck {
+    task: Option<HealthcheckTask>,
+    healthcheck_address: String,
+    handle: Option<JoinHandle<()>>,
+}
+
+impl Healthcheck {
+    pub async fn create(
+        healthcheck_address: &str,
+        udp: &UdpTransport,
+        udp_socket_address: SocketAddr,
+    ) -> Result<Self> {
+        let peer = if udp_socket_address.ip().is_unspecified() {
+            HostnamePort::new("localhost", udp_socket_address.port()).to_string()
+        } else {
+            udp_socket_address.to_string()
+        };
+
+        let ctx = udp.ctx().async_try_clone().await?;
+
+        let task = HealthcheckTask {
+            ctx,
+            udp: udp.clone(),
+            peer,
+        };
+
+        Ok(Self {
+            task: Some(task),
+            healthcheck_address: healthcheck_address.to_string(),
+            handle: None,
+        })
+    }
+
+    pub async fn start(&mut self) -> Result<()> {
+        if self.handle.is_some() {
+            return Err(Error::new(
+                Origin::Application,
+                Kind::Unknown,
+                "Can't start Healthcheck because handle is present",
+            ));
+        }
+
+        let task = self.task.take().ok_or(Error::new(
+            Origin::Application,
+            Kind::Unknown,
+            "Can't start Healthcheck because handle is present",
```

Comment:
Shouldn't this say something about the task?

Reply:
Same here